src/include/ipc4/base_fw.h

```diff
@@ -0,0 +1,641 @@
+	/* Configuration of clocks */
+	IPC4_CLOCKS_CONFIGURATION               = 20,
+	/* USB Audio Offload support */
+	IPC4_UAOL_SUPPORT                       = 21,
+	/* Configuration of Dynamic Power Gating Policy */
+	IPC4_POWER_GATING_POLICY                = 22,
+	/* Configuration of assert mode
+	 * Run-time asserts requires special handling by decoder.
+	 * Asserts will be in format: "%passert", ptr_to_assert_desc.
+	 *
+	 * ptr_to_assert_desc will point to place in .asserts_desc section in ELF.
+	 * Data must be cased to struct: struct assert_entry
+	 */
+	IPC4_ASSERT_MODE                        = 23,
+	/* Size of telemetry buffer in bytes. The default size is 4KB  */
+	IPC4_TELEMETRY_BUFFER_SIZE = 24,
+	/* HW version information  */
+	IPC4_BUS_HARDWARE_ID = 25,
+	/* Total number of FW config parameters  */
+	IPC4_FW_CFG_PARAMS_COUNT,
+	/* Max config parameter id */
+	IPC4_MAX_FW_CFG_PARAM = IPC4_FW_CFG_PARAMS_COUNT - 1,
+};
+
+enum ipc4_hw_config_params {
+	/* Version of cAVS implemented by FW (from ROMInfo) */
+	IPC4_CAVS_VER_HW_CFG             = 0,
+	/* How many dsp cores are available in current audio subsystem */
+	IPC4_DSP_CORES_HW_CFG            = 1,
+	/* Size of a single memory page */
+	IPC4_MEM_PAGE_BYTES_HW_CFG       = 2,
+	/* Total number of physical pages available for allocation */
+	IPC4_TOTAL_PHYS_MEM_PAGES_HW_CFG = 3,
+	/*
+	 * Number of items in controller_base_addr array is specified by
+	 * controller_count. note Lower 16 bits of I2sVersion may be used
+	 * in future to indicate implementation revision if necessary.
+	 */
+	IPC4_I2S_CAPS_HW_CFG             = 4,
+	/* GPDMA capabilities */
+	IPC4_GPDMA_CAPS_HW_CFG           = 5,
+	/*Total number of DMA gateways of all types */
+	IPC4_GATEWAY_COUNT_HW_CFG        = 6,
+	/* Number of HP SRAM memory banks manageable by DSP */
+	IPC4_HP_EBB_COUNT_HW_CFG          = 7,
+	/* Number of LP SRAM memory banks manageable by DSP */
+	IPC4_LP_EBB_COUNT_HW_CFG          = 8,
+	/* Size of a single memory bank (EBB) in bytes */
+	IPC4_EBB_SIZE_BYTES_HW_CFG        = 9,
+	/* UAOL capabilities */
+	IPC4_UAOL_CAPS_HW_CFG            = 10
+};
+
+struct ipc4_tuple {
+	uint32_t type;
+	uint32_t length;
+	char data[0];
+};
+
+enum ipc4_memory_type {
+	/* High power sram memory */
+	IPC4_HP_SRAM_MEMORY = 0,
+	/*Low power sram memory */
+	IPC4_LP_SRAM_MEMORY = 1
+};
+
+enum ipc4_resource_state_request {
+	/*
+	 * This type is used to inform about free physical HP sram memory pages
+	 * available.
+	 */
+	IPC4_FREE_PHYS_MEM_PAGES = 0
+};
+
+/* PhysMemPages describes current free phys memory pages */
+struct ipc4_phys_mem_pages {
+	uint32_t mem_type;
+	/* Number of pages */
+	uint32_t pages;
+};
+
+enum ipc4_dx_states {
+	/*D0 state means that dsp in this state is powered up and ready for processing */
+	IPC4_D0_STATE = 0,
+	/* D3 state means that dsp in this state is powered down or not yet
+	 * ready for processing. No idc/ipc should be send to core in this state.
+	 */
+	IPC4_D3_STATE = 3
```

Comment:
D0i3 ?

Reply:
according to comments " powered down", so it should be D3

Reply:
ok, can we have a comment here for D0i3 i.e. "TODO: how do we enter D0i3"

Reply:
sure

Reply:
D0i3 is a sub-state of D0